examples/pytorch/MNIST/example1/mnist_autolog_example1.py

```diff
@@ -278,13 +278,16 @@ def configure_optimizers(self):
     # Add trainer specific arguments
 
     parser.add_argument(
-        "--max-epochs", type=int, default=20, help="number of epochs to run (default: 20)"
+        "--max_epochs", type=int, default=20, help="number of epochs to run (default: 20)"
```

Comment:
@shrinath-suresh Can we revert this change?

Reply:
@harupy This is the breaking change. The number of epochs parameter is set to `max_epochs` in read me files and MLProject and still it was  `max-epochs` in script.

Reply:
Can we fix README instead?

Reply:
@harupy The name changes were done per previous request from @smurching Can you please check with him on what we want as the final naming scheme. For `mlflow run .` the underscore and hypens are not interchangeable

Reply:
Sorry for the confusion, I think it's fine to be consistent in our usage of ``--max-epochs`` across the script and MLproject file. My point in the other PR was more that we should be consistent in naming a parameter `max-epochs` in both places rather than `epochs` in one and `max-epochs` elsewhere, I didn't mean to express a preference in favor of underscores

In general, the POSIX convention is to use hyphens (`-`) rather than underscores (`_`) in parameter names: https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html, so seems reasonable to follow the same convention here, and use hyphens in both the script & MLproject file as @harupy is suggesting

Reply:
@smurching @harupy @chauhang Pytorch Lightning uses `_` in argument parser (Ex: `max_epochs`). As @SeanNaren suggested in https://github.com/mlflow/mlflow/pull/3655#issuecomment-724222640 I have removed all the trainer specific arguments and used PTL's argument options. Hence, we cannot pass `max-epochs`anymore with the current code. 

To maintain the consistency, i will update the variables to have `_` (Ex: `batch_size` , `num_workers`).

Reply:
@shrinath-suresh 

> we cannot pass `max-epochs` anymore with the current code.

Does this mean 

```
python xxx.py --max-epochs 3
```

Reply:
Yes you are right 

```
python bert_classification.py --max-epochs 1

usage: bert_classification.py [-h] [--num-samples N] [--logger [LOGGER]] [--checkpoint_callback [CHECKPOINT_CALLBACK]] [--default_root_dir DEFAULT_ROOT_DIR] [--gradient_clip_val GRADIENT_CLIP_VAL]
                              [--process_position PROCESS_POSITION] [--num_nodes NUM_NODES] [--num_processes NUM_PROCESSES] [--gpus GPUS] [--auto_select_gpus [AUTO_SELECT_GPUS]] [--tpu_cores TPU_CORES]
                              [--log_gpu_memory LOG_GPU_MEMORY] [--progress_bar_refresh_rate PROGRESS_BAR_REFRESH_RATE] [--overfit_batches OVERFIT_BATCHES] [--track_grad_norm TRACK_GRAD_NORM]
                              [--check_val_every_n_epoch CHECK_VAL_EVERY_N_EPOCH] [--fast_dev_run [FAST_DEV_RUN]] [--accumulate_grad_batches ACCUMULATE_GRAD_BATCHES] [--max_epochs MAX_EPOCHS]
                              [--min_epochs MIN_EPOCHS] [--max_steps MAX_STEPS] [--min_steps MIN_STEPS] [--limit_train_batches LIMIT_TRAIN_BATCHES] [--limit_val_batches LIMIT_VAL_BATCHES]
                              [--limit_test_batches LIMIT_TEST_BATCHES] [--val_check_interval VAL_CHECK_INTERVAL] [--flush_logs_every_n_steps FLUSH_LOGS_EVERY_N_STEPS]
                              [--log_every_n_steps LOG_EVERY_N_STEPS] [--accelerator ACCELERATOR] [--sync_batchnorm [SYNC_BATCHNORM]] [--precision PRECISION] [--weights_summary WEIGHTS_SUMMARY]
                              [--weights_save_path WEIGHTS_SAVE_PATH] [--num_sanity_val_steps NUM_SANITY_VAL_STEPS] [--truncated_bptt_steps TRUNCATED_BPTT_STEPS]
                              [--resume_from_checkpoint RESUME_FROM_CHECKPOINT] [--profiler [PROFILER]] [--benchmark [BENCHMARK]] [--deterministic [DETERMINISTIC]]
                              [--reload_dataloaders_every_epoch [RELOAD_DATALOADERS_EVERY_EPOCH]] [--auto_lr_find [AUTO_LR_FIND]] [--replace_sampler_ddp [REPLACE_SAMPLER_DDP]]
                              [--terminate_on_nan [TERMINATE_ON_NAN]] [--auto_scale_batch_size [AUTO_SCALE_BATCH_SIZE]] [--prepare_data_per_node [PREPARE_DATA_PER_NODE]] [--amp_backend AMP_BACKEND]
                              [--amp_level AMP_LEVEL] [--distributed_backend DISTRIBUTED_BACKEND] [--automatic_optimization [AUTOMATIC_OPTIMIZATION]] [--lr LR] [--batch-size N] [--num-workers N]
bert_classification.py: error: unrecognized arguments: --max-epochs 1

```

Reply:
do we really need these options?

Reply:
The big list is from Pytorch Lightning's argument list. Not from the script.

Reply:
@harupy In the MLProject files, we have kept the relevant parameters for ease of reproducibility. All the others mentioned above are the optional prams that PTL supports